clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1476,9 +1497,15 @@ public void handleResponse(AbstractResponse response) {
                 } else if (error == Errors.GROUP_AUTHORIZATION_FAILED) {
                     abortableError(GroupAuthorizationException.forGroupId(builder.data.groupId()));
                     break;
-                } else if (error == Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED
-                        || error == Errors.INVALID_PRODUCER_EPOCH
-                        || error == Errors.UNSUPPORTED_FOR_MESSAGE_FORMAT) {
+                } else if (isGroupFencingException(error)) {
+                    if (enableGroupFencing) {
```

Comment:
We will throw IllegalStateException if unexpected group fencing exception was thrown for old API

Reply:
I think this check is overkill.

Reply:
But theoretically possible?

Reply:
After offline discussion, this shall be removed for simplicity.